src/components/Documentation/FieldLink/FieldLink.tsx

```diff
@@ -0,0 +1,39 @@
+import { GraphQLType } from 'graphql';
+import { useContext } from 'react';
+import { DocumentationContext } from '../../../context/documentationContext';
+import type { FieldType } from '../../../types/documentationTypes';
+import classes from './field-link.module.scss';
+
+type FieldLinkProps = {
+  field: FieldType;
+  type: GraphQLType;
+};
+
+const FieldLink = ({ field, type }: FieldLinkProps) => {
+  const { setStack } = useContext(DocumentationContext);
+  return (
+    <a
+      href="#"
+      className={classes.field}
+      onClick={(e) => {
+        e.preventDefault();
+        setStack((curr) => {
+          return [
+            ...curr,
+            {
+              name: field.name,
+              view: 'field',
+              args: field.args,
+              description: field.description,
+              type,
+            },
+          ];
+        });
+      }}
```

Comment:
would be great if you splitted this handler into the body of component. 
consider using destructing as well